src/components/package-show/package-show.js

```diff
@@ -138,15 +139,26 @@ export default class PackageShow extends Component {
                   id="package-details-toggle-switch"
                 />
               </label>
-
-              {model.visibilityData.isHidden && (
-                <div data-test-eholdings-package-details-is-hidden>
-                  <p><strong>This package is hidden.</strong></p>
-                  <p><em>{model.visibilityData.reason}</em></p>
-                  <hr />
+              <label
+                data-test-eholdings-package-details-hidden
+                htmlFor="package-details-toggle-hidden-switch"
+              >
+                <h4>{model.visibilityData.isHidden ? 'Hide' : 'Show'}</h4>
+                <div className={styles['flex-container']}>
+                  <div className={styles['flex-item']}>
+                    <ToggleSwitch
+                      onChange={this.props.toggleHidden}
+                      checked={model.visibilityData.isHidden}
+                      isPending={model.update.isPending}
```

Comment:
Just merged an improvement for this type of thing today: https://github.com/folio-org/ui-eholdings/pull/162.

Now you can say `isPending={model.update.isPending && 'visibilityData' in model.update.changedAttributes}`. That isolates the loading states for the selection toggle and the visibility toggle, so they're not pulsing in a loading state together.

Reply:
Thanks for suggestions - will test both of these out!